enterprise_gateway/services/processproxies/yarn.py

```diff
@@ -37,22 +37,27 @@ class YarnClusterProcessProxy(RemoteProcessProxy):
     def __init__(self, kernel_manager, proxy_config):
         super(YarnClusterProcessProxy, self).__init__(kernel_manager, proxy_config)
         self.application_id = None
+        self.rm_addr = None
         self.yarn_endpoint \
             = proxy_config.get('yarn_endpoint',
                                kernel_manager.parent.parent.yarn_endpoint)
         self.yarn_endpoint_security_enabled \
             = proxy_config.get('yarn_endpoint_security_enabled',
                                kernel_manager.parent.parent.yarn_endpoint_security_enabled)
-        yarn_url = urlparse(self.yarn_endpoint)
-        yarn_master = yarn_url.hostname
-        yarn_port = yarn_url.port
-        if self.yarn_endpoint_security_enabled is True:
-            self.resource_mgr = ResourceManager(address=yarn_master,
-                                                port=yarn_port,
-                                                kerberos_enabled=self.yarn_endpoint_security_enabled)
-        else:
-            self.resource_mgr = ResourceManager(address=yarn_master,
-                                                port=yarn_port)
+
+        yarn_master = None
+        yarn_port = 8088
+        if self.yarn_endpoint:
+            yarn_url = urlparse(self.yarn_endpoint)
+            yarn_master = yarn_url.hostname
+            yarn_port = yarn_url.port
```

Comment:
It seems that when yarn_master is none, the port will never be used. In this case, should we actually initialize the yarn_port as None?

Reply:
It doesn't matter since it will be pulled from the config. I was following the defaulting used in yarn client. That said it probably makes more sense to follow suit and pass none in these cases. This will be okay as long as yarn client only uses the port parameter when the address parameter is not none. I'll change this tomorrow.

Reply:
@lresende - update pushed.